wire-ios/Wire-iOS/Sources/UserInterface/DeviceView/DeviceInfoViewModel.swift

```diff
@@ -23,129 +23,104 @@ import WireDataModel
 import WireSyncEngine
 
 protocol DeviceDetailsViewActions {
-    var isMLSEnabled: Bool { get }
-    var isE2eIdentityEnabled: Bool { get }
     var isSelfClient: Bool { get }
     var isProcessing: ((Bool) -> Void)? { get set }
 
-    func fetchCertificate() async -> E2eIdentityCertificate?
-    func fetchMLSThumbprint() async -> String?
+    func getCertificate() async -> E2eIdentityCertificate?
+    func enrollClient() async -> E2eIdentityCertificate?
+    func updateCertificate() async -> E2eIdentityCertificate?
     func removeDevice() async -> Bool
     func resetSession()
     func updateVerified(_ value: Bool) async -> Bool
     func copyToClipboard(_ value: String)
     func downloadE2EIdentityCertificate(certificate: E2eIdentityCertificate)
+    func isE2eIdentityEnabled() async -> Bool
+    func getProteusFingerPrint() async -> String
 }
 
 final class DeviceInfoViewModel: ObservableObject {
-    let userSession: UserSession
     let addedDate: String
     let proteusID: String
-    let getUserClientFingerprint: GetUserClientFingerprintUseCaseProtocol
     let userClient: UserClient
-    var title: String
+    let gracePeriod: TimeInterval
 
+    var title: String
     var isSelfClient: Bool
 
-    var isE2EIdentityEnabled: Bool {
-        actionsHandler.isE2eIdentityEnabled
-    }
-
-    var isMLSEnablled: Bool {
-        actionsHandler.isMLSEnabled
-    }
-
-    var isValidCerificate: Bool {
-        guard let certificate = e2eIdentityCertificate,
-           E2EIdentityCertificateStatus.status(for: certificate.certificateStatus) != .none,
-              E2EIdentityCertificateStatus.status(for: certificate.certificateStatus) != .notActivated else {
-            return false
-        }
-        return true
+    var isCopyEnabled: Bool {
+        return Settings.isClipboardEnabled
     }
 
-    var certificateStatus: E2EIdentityCertificateStatus {
-        guard let certificate = e2eIdentityCertificate,
-              let status = E2EIdentityCertificateStatus.allCases.filter({
-                        $0.titleForStatus() == certificate.certificateStatus
-                    }
-                ).first
-        else {
-            return isE2EIdentityEnabled ? .notActivated : .none
+    var isCertificateExpiringSoon: Bool? {
+        guard let certificate = e2eIdentityCertificate else {
+            return nil
         }
-        return status
+        return certificate.shouldCertificateBeUpdated(with: gracePeriod)
     }
 
-    var isCertificateExpiringSoon: Bool {
-        guard let certificate = e2eIdentityCertificate else {
-            return false
-        }
-        return certificate.expiryDate < Date.now + .oneDay + .oneDay
+    var mlsThumbprint: String? {
+        (userClient.mlsPublicKeys.ed25519 ?? e2eIdentityCertificate?.mlsThumbprint)?
```

Comment:
Are they different? 
**suggestion:** to move it to the separate property

Reply:
They should be same. Looking at the description of it. If you do not have `userClient.mlsPublicKeys.ed25519 ` then ideally your client is not enrolled which means Not activated.

Reply:
So we only need use `userClient.mlsPublicKeys.ed25519` then?

Reply:
Yeah, we could do that. Just confirmed it with Mojtaba